### PR TITLE
Support vlan protocol

### PR DESCRIPTION
**Pull request by soyking**
Support get vlan with protocol and create vlan with protocol, see #429 

Reference: 
- https://elixir.bootlin.com/linux/v4.4/source/net/8021q/vlan_netlink.c#L57
- https://elixir.bootlin.com/linux/v4.4/source/include/uapi/linux/if_ether.h#L65

### CODE REVIEW

**Comment by soyking** [comment]
@vishvananda Could you please have a look for this PR ? ><

File: link.go [hunk 1]
```diff
@@ -405,6 +406,43 @@ func (ipvlan *IPVlan) Type() string {
...
+
+func (p VlanProtocol) String() string {
+	s, ok := VlanProtocolToString[p]
+	if !ok {
+		return fmt.Sprintf("VlanProtocol(%d)", p)
+	}
+	return s
+}
+
+// StringToVlanProtocol returns vlan protocol, or uknonw is the s is invalid.
```
Comment (jellonek) [inline link.go]: Typo: unknown

File: link.go [hunk 2]
```diff
@@ -405,6 +406,43 @@ func (ipvlan *IPVlan) Type() string {
...
+func StringToVlanProtocol(s string) VlanProtocol {
+	mode, ok := StringToVlanProtocolMap[s]
+	if !ok {
+		return VLAN_PROTOCOL_UNKNOWN
+	}
+	return mode
+}
+
+// Possible VlanProtocol
```
Comment (jellonek) [inline link.go]: VlanProtocol possible values

File: link.go (hunk 1 above)
Comment (soyking) [reply link.go]: Fixed

File: link.go (hunk 2 above)
Comment (soyking) [reply link.go]: Fixed

**Comment by vishvananda** [comment]
This LGTM. Can you squash it into one commit, please?

**Comment by soyking** [comment]
> This LGTM. Can you squash it into one commit, please?

Done

**Pull request merged**